py-polars/polars/lazyframe/frame.py

```diff
@@ -3181,7 +3180,6 @@ def with_columns(
 
         return self._from_pyldf(self._ldf.with_columns(pyexprs))
 
-    @typing.no_type_check
     def with_context(self, other: Self | list[Self]) -> Self:
```

Comment:
Not sure why this needed a `no_type_check` decorator - I get no issues without it.